chaosazure/vmss/actions.py

```diff
@@ -242,8 +247,21 @@ def choose_vmss_instance_at_random(vmss_choice, configuration, secrets):
         logger.debug(
             "Found virtual machine scale set instances: {}".format(
                 [x['name'] for x in vmss_instances]))
-    logger.warning("Temporarily hard coding to the first instance which sould be zone 0")            
-    choice_vmss_instance = vmss_instances[0]
+    logger.warning("Temporarily hard coding to the first instance which sould be zone 0")
+    choice_vmss_instance = random.choice(vmss_instance)
+    return choice_vmss_instance
+
+def choose_vmss_instance(vmss_choice, configuration, secrets, vmss_name):
+    vmss_instances = fetch_vmss_instances(vmss_choice, configuration, secrets)
+    if not vmss_instances:
+        logger.warning("No virtual machine scale set instances found")
+        raise FailedActivity("No virtual machine scale set instances found")
+    else:
+        logger.debug(
+            "Found virtual machine scale set instances: {}".format(
+                [x['name'] for x in vmss_instances]))
+    logger.warning("Attempting to stop instance with name %s" % vmss_name)
+    choice_vmss_instance = next(vmss for vmss in vmss_instances if vmss['name'] == vmss_name)
```

Comment:
Interesting approach. It's cunning actually :)

We have a mojo in the chaostoolkit code base, it must remain readbable to folks who don't know the Python language whenever we can. Here, I'm happy with:

```
for vmss in vmss_instances:
    if vmss['name'] == vmss_name:
       return vmss
```

Boring and oldschool but easier to read IMO.

Reply:
I think I wound up doing something more similar for the final approach anyway!